hal/ticker_api.h

```diff
@@ -73,6 +73,7 @@ typedef struct {
     ticker_event_handler event_handler; /**< Event handler */
     ticker_event_t *head;               /**< A pointer to head */
     uint32_t frequency;                 /**< Frequency of the timer in Hz */
+    uint32_t frequency_shifts;          /**< Frequency shift if divisible by 2, ohterwise 0 */
```

Comment:
`otherwise`. Really no need to have a `uint32_t` for this - suggest making it a `uint8_t` and popping it next to initialised. Might squeeze it in and avoid a binary compatibility breakage.

Reply:
Put it after initialised, so it slots into what should presumably be 3 bytes of padding at the end. We have a number of binary components, and if we can avoid upsetting them by modifying these central structures, it's best.